src/NBitcoin/ScriptReader.cs

```diff
@@ -748,9 +748,12 @@ public IEnumerable<Op> ToEnumerable()
         {
             Op code;
             while((code = Read()) != null)
-            {
                 yield return code;
-            }
+        }
+
+        public void Dispose()
+        {
+            this._Inner.Dispose();
```

Comment:
what happens if the object is null?

Reply:
Hmm. 
I assumed that check in the constructor would be enough. 
But it is possible to set it to null after the creation so I'll add a null check. 
Thx for pointing that out.